File: content/projects/build-shell/starters/javascript/02-read-input/main.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
});

process.stdout.write("$ ");

rl.on("line", (line) => {
  // TODO: Handle the input line
  // TODO: For now, print "<input>: command not found" for any input
  // TODO: Then print the prompt again
  //
  // Hints:
  //   - Use process.stdout.write() to print without a trailing newline
  //   - Print the "command not found" message:
  //     process.stdout.write(`${line}: command not found\n`)
  //   - After printing, show the prompt again with process.stdout.write('$ ')

  process.stdout.write("$ ");
});

rl.on("close", () => {
  process.exit(0);
});

File: content/projects/build-shell/starters/python/02-read-input/main.py
import sys


def main():
    while True:
        try:
            print("$ ", end="", flush=True)
            line = input()

            # TODO: Handle the input line
            # TODO: For now, print "<input>: command not found" for any input
            #
            # Hints:
            #   - The variable `line` contains what the user typed (without newline)
            #   - Print the "command not found" message using an f-string:
            #     print(f"{line}: command not found")
            #   - Don't forget to handle empty input (just pressing Enter)

        except EOFError:
            break


if __name__ == "__main__":
    main()
